ibis_tpc/runners.py

```diff
@@ -278,29 +278,72 @@ def compare(rows1, rows2):
         for k in keys:
             v1 = lcr1.get(k, None)
             v2 = lcr2.get(k, None)
-            if isinstance(v2, pandas.Timestamp):
-                if v1 != v2.strftime("%Y-%m-%d"):
-                    diffs.append(f"[{i}].{k} (date) {v1} != {v2}")
-            elif isinstance(v1, float) and isinstance(v2, float):
-                if v2 != v1:
-                    if v1:
-                        dv = abs(v2 - v1)
-                        pd = dv / v1
-                    else:
-                        pd = 1
-                    if pd > 1e-10:
-                        diffs.append(f"[{i}].{k} (float) {v1} != {v2} ({pd*100}%)")
-            elif isinstance(v1, float) and math.isnan(v1) and v2 is None:
-                pass
-            elif isinstance(v2, float) and math.isnan(v2) and v1 is None:
-                pass
-            else:
-                if v1 != v2:
-                    diffs.append(f"[{i}].{k} {v1} ({type(v1)}) != {v2} ({type(v2)})")
+            _compare(v1, v2, row=row, key=k, diffs=diffs)
 
     return diffs
 
 
+@dispatch(pandas.Timestamp, pandas.Timestamp)
+def _compare(v1, v2, row=None, key=None, diffs=None):
+    if v1 != v2:
+        diffs.append(f"[{row}].{key} (date) {v1} != {v2}")
```

Comment:
How about `return` or `yield` instead of passing in the container itself?

Reply:
Yep, that'll be much cleaner